MoreLinq.Test/IgnoreErrorsTest.cs

```diff
@@ -0,0 +1,92 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2018 Leandro F. Vieira (leandromoh). All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using NUnit.Framework;
+    using System;
+
+    [TestFixture]
+    public class IgnoreErrorsTest
+    {
+        [Test]
+        public void IgnoreErrorsIsLazy()
+        {
+            new BreakingSequence<int>().IgnoreErrors(BreakingFunc.Of<Exception, bool>());
+        }
+
+        [Test]
+        public void IgnoreErrors()
+        {
+            var source = MoreEnumerable.From(() => 1,
+                                             () => throw new TestException("True"),
+                                             () => 2,
+                                             () => throw new TestException("False"),
+                                             () => 3);
+
+            var result = source.IgnoreErrors((TestException e) => bool.Parse(e.Message));
```

Comment:
This predicate looks strange. Why not simply use [`Exception.Data`](https://docs.microsoft.com/en-us/dotnet/api/system.exception.data#System_Exception_Data) instead, like below?

```c#
var key = new object();
var source = MoreEnumerable.From(() => 1,
                                 () => throw new TestException { Data = { [key] = true } },
                                 () => 2,
                                 () => throw new TestException { Data = { [key] = false } },
                                 () => 3);

var result = source.IgnoreErrors((TestException e) => (bool) e.Data[key]);
```